Moller-Plesset/LT-MP2.py

```diff
@@ -0,0 +1,100 @@
+"""
+A reference implementation of MP2 using the Laplace transformation for a restricted reference.
+This is performed in an MO basis for demonstration, but LT-MP2 can be extended to AO-LT-MP2.
+
+References:
+    M. Haser, J. Almlof, and G. E. Scuseria, Chem. Phys. Lett. 181, 497 (1991).
+    P. Y. Ayala and G. E. Scuseria, J. Chem. Phys., 110, 3660 (1999).
+"""
+
+__authors__ = "Oliver J. Backhouse"
+__credits__ = ["Oliver J. Backhouse"]
+
+__copyright__ = "(c) 2014-2020, The Psi4NumPy Developers"
+__license__ = "BSD-3-Clause"
+__date__ = "2018-03-01"
+
+import time
+import numpy as np
+import psi4
+
+# Settings
+compare_to_psi4 = True
+grid_size = 40
+
+# Set the memory and output file
+psi4.set_memory('2 GB')
+psi4.core.set_output_file('output.dat', False)
+
+# Set molecule and basis
+mol = psi4.geometry("""
+O
+H 1 1.1
+H 1 1.1 2 104
+symmetry c1
+""")
+
+psi4.set_options({'basis': 'aug-cc-pvdz'})
+
+# Perform SCF
+print('\nPerforming SCF...')
+e_scf, wfn = psi4.energy('SCF', return_wfn=True)
+mints = psi4.core.MintsHelper(wfn.basisset())
+
+# Get occupied and virtual MO energies and coefficients
+e_occ = wfn.epsilon_a_subset('AO', 'ACTIVE_OCC').np 
+e_vir = wfn.epsilon_a_subset('AO', 'ACTIVE_VIR').np
+c_occ = wfn.Ca_subset('AO', 'OCC')
+c_vir = wfn.Ca_subset('AO', 'VIR')
+
+# Get the two-electron integrals in MO basis
+print('Building MO integrals...')
+iajb = mints.mo_eri(c_occ, c_vir, c_occ, c_vir).np
+
+# Build the grids and weights according to Gauss-Laguerre quadrature
+# Also apply a weighting function w(x) = exp(x)
+grid, weights = np.polynomial.laguerre.laggauss(grid_size)
+weights *= np.exp(grid)
+
+# Loop over grid points and compute energies
+e_mp2_corr_os = 0.0
+e_mp2_corr_ss = 0.0
+
+print('Looping over %d grid points...' % grid_size)
+t_start = time.time()
+for t, w in zip(grid, weights):
+    # Transform the two-electron integrals into the time-dependent form
+    t_occ = np.exp( t * 0.5 * e_occ)
+    t_vir = np.exp(-t * 0.5 * e_vir)
+    iajb_t = np.einsum('p,q,r,s,pqrs->pqrs', t_occ, t_vir, t_occ, t_vir, iajb)
```

Comment:
Point out that this is directly building the amplitudes. Often in other equations we form and contract the amplitudes in the same term.

Reply:
Sure - thinking about this, it would probably make more physical sense building the amplitudes without the factor 0.5, and then contracting these with the MO integrals.